### DISCUSSION TITLE
【数据安全】Use Notion integrations to read databases

### DISCUSSION THREAD

**Discussion by 1208nn** [tangly1024/NotionNext#2124] [Ideas]
**为什么提出这个新的特性改动**
Prefer not to make the database public and not to update token frequently.

**描述一下你推荐的解决方案**
<https://elog.1874.cool/notion/gvnxobqogetukays#notion>

**描述一下你考虑过的其它替代解决方案**
Use account and password to fetch from notion simulating a user.

**补充说明**
[integrations](https://www.notion.so/my-integrations)
[Docs](https://developers.notion.com/docs/getting-started#step-1-create-an-integration)

**Comment by alex-guoba** [comment]
https://github.com/NotionX/react-notion-x?tab=readme-ov-file#private-pages

请问有切换到官方sdk的计划吗？

**Reply by alex-guoba** [reply]
自己实现了一个，公共API有些能力上不支持，比如Database Gallery View。估计没办法平滑切换

**Reply by ChenyqThu** [reply]
是的，目前有这个问题，API不支持获取 formula/database view之类的，我在想能不能两者结合起来？

**Reply by alex-guoba** [reply]
现在 `react-notion-x` 依赖的API没有官方文档，也没有后续维护计划。
如果不能解除现有依赖的话，那切换意义就不大了吧。

**Reply by ChenyqThu** [reply]
react-notion-x用的还是非官方的前段破解API，官方API似乎react-notion-x并没有支持。

**Reply by alex-guoba** [reply]
是的，所以建议别用了
